File: 01_DNA_counting_nucleotides/R/01_counting_nucleotides.R
#!/usr/bin/env Rscript

input <- readLines('stdin', n=1, warn=FALSE)

countBases <- function(base, seq) {
  # Count the occurences of a character or substring in a string
  # 
  # Args:
  #   base: A character, string or regex to be searched for
  #   seq:  The string to be searched in.
  #
  # Returns:
  #   The number of occurences of base in seq.
  n <- length(attr(gregexpr(pattern=base, text=seq)[[1]], "match.length"))
  return(n)
}

output <- list()

output[['A']] <- countBases("A", input)
output[['C']] <- countBases("C", input)
output[['G']] <- countBases("G", input)
output[['T']] <- countBases("T", input)

cat(c(output$A, output$C, output$G, output$T, '\n'))

File: 01_DNA_counting_nucleotides/julia/01_counting_nucleotides.jl
using Match

s = ""

count = {"a" => 0, "c" => 0, "g" => 0, "t" => 0}

s = readline(STDIN)

for i = 1:endof(s)
  @match s[i] begin
    'A' => count["a"] += 1
    'C' => count["c"] += 1
    'G' => count["g"] += 1
    'T' => count["t"] += 1
     _  => continue
  end
end

@printf("%d %d %d %d\n", count["a"], count["c"], count["g"], count["t"])